### ISSUE TITLE
Grid Map throws 'Invalid attempt to destructure non-iterable instance'

### ISSUE THREAD

**Issue by nkconnor** [metabase/metabase#9020]
Grid Map throws console error under certain circumstances

- Your browser and the version: Chrome Version 70.0.3538.77 (Official Build) (64-bit)
- Your operating system: Ubuntu 16.04
- Your databases: MySQL
- Metabase version: You're on version v0.31.1
- Metabase hosting environment: Linux 18.04
- Metabase internal database: H2

**Browser errror**
```
TypeError: Invalid attempt to destructure non-iterable instance
    at app-main.bundle.js?3bdf670dc92c77224541:5
    at t.value (app-main.bundle.js?3bdf670dc92c77224541:5)
    at t.value (app-main.bundle.js?3bdf670dc92c77224541:5)
    at t.notifyAll (vendor.bundle.js?3bdf670dc92c77224541:139)
    at h.close (vendor.bundle.js?3bdf670dc92c77224541:145)
    at h.closeAll (vendor.bundle.js?3bdf670dc92c77224541:94)
    at h.perform (vendor.bundle.js?3bdf670dc92c77224541:94)
    at y.perform (vendor.bundle.js?3bdf670dc92c77224541:94)
    at y.perform (vendor.bundle.js?3bdf670dc92c77224541:83)
    at Object.w [as flushBatchedUpdates] (vendor.bundle.js?3bdf670dc92c77224541:83)
....
```

I haven't yet narrowed down when the error occurs (e.g. exceeding some # of rows or # of grids)

**Comment by nkconnor** [comment]
This is caused by a `NULL` latitude/longitude value - not sure if this should just be closed - or perhaps turned into a feature request for better error message?

**Comment by salsakran** [comment]
we should definitely have a better error message.

**Comment by ohld** [comment]
It is impossible now to filter LATTITUDE IS NOT NULL because since this column has the 'geo' type it is impossible to apply filter "IS SET" on it.

The hack that helped: add CUSTOM EXPRESSION filter with `LATTITUDE > -180`.

**Comment by flamber** [comment]
@ohld I think you're looking for #7361 and #12225.

**Comment by flamber** [comment]
Fixed by #12159
